eth/query/block.go

```diff
@@ -212,6 +212,14 @@ func GetTxObjectFromBlockResult(
 	}
 	txObj.Input = eth.EncBytes(input)
 
+	evmTxHash, err := eth.DecDataToBytes(txObj.Hash)
+	if err != nil {
+		return eth.GetEmptyTxObject(), nil, err
+	}
+	if evmAuxStore.IsDupEVMTxHash(evmTxHash) {
```

Comment:
This should just run for EVM txs, and if you move this up you won't have to decode it.

Reply:
Done